agent/internal/task_config.go

```diff
@@ -111,39 +131,21 @@ func (c *TaskConfig) GetCloneMethod() string {
 	return evergreen.CloneMethodOAuth
 }
 
-// GetTaskGroup returns the task group for the given task group name. It may
-// return nil if the task group name is empty.
-func (tc *TaskConfig) GetTaskGroup(taskGroup string) (*model.TaskGroup, error) {
-	if err := tc.Validate(); err != nil {
-		return nil, err
-	}
-
-	if taskGroup == "" {
-		return nil, nil
-	}
-	tg := tc.Project.FindTaskGroup(taskGroup)
-	if tg == nil {
-		return nil, errors.Errorf("couldn't find task group '%s' in project '%s'", taskGroup, tc.Project.Identifier)
-	}
-
-	return tg, nil
-}
-
 // Validate validates that the task config is populated with the data required
 // for a task to run.
+// Note that this is here only as legacy code. These checks are not sufficient
```

Comment:
do we want to remove these calls then? Or shore up the validate function as future work?

Reply:
We could remove these (they seem to only apply to tests that don't load the task config from the app server), but I didn't want to increase the amount of changes made by this PR. We can have a follow-up to delete the validation.

Reply:
That works for me!

Reply:
I forgot to mention, but the ticket to get rid of the extra validation is EVG-20852.